scripts/utils/tensor_utils.dml

```diff
@@ -0,0 +1,73 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+/*
+ * Reshape util for tensors in NCHW format.
+ * Transposes the 1st and 2nd dimensions.
+ */
+reshape_NCHW_to_CNHW = function(matrix[double] X, int C) return (matrix[double] out){
```

Comment:
Can we rename this to `transpose_NCHW_to_CNHW`?  The reasoning is that NumPy has a [`reshape`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.reshape.h) and [`transpose`](https://docs.scipy.org/doc/numpy-1.12.0/reference/generated/numpy.transpose.html) function.  The former just changes the shape of the tensor, without moving any of the elements.  This would be equivalent to our `matrix(X, rows=a, cols=b)` for the 2D case.  The latter function permutes the axes as desired, which would be equivalent to this new function you've added.